crates/router/src/core/payment_methods/migration.rs

```diff
@@ -149,3 +149,64 @@ fn validate_card_exp_year(year: String) -> Result<(), errors::ValidationError> {
         })
     }
 }
+
+#[derive(Debug)]
+pub struct RecordMigrationStatus {
+    pub card_migrated: Option<bool>,
+    pub network_token_migrated: Option<bool>,
+    pub connector_mandate_details_migrated: Option<bool>,
+    pub network_transaction_migrated: Option<bool>,
+}
+
+#[derive(Debug)]
+pub struct RecordMigrationStatusBuilder {
+    pub card_migrated: Option<bool>,
+    pub network_token_migrated: Option<bool>,
+    pub connector_mandate_details_migrated: Option<bool>,
+    pub network_transaction_migrated: Option<bool>,
+}
+
+impl RecordMigrationStatusBuilder {
+    pub fn new() -> Self {
+        Self {
+            card_migrated: None,
+            network_token_migrated: None,
+            connector_mandate_details_migrated: None,
+            network_transaction_migrated: None,
+        }
+    }
+
+    pub fn card_migrated(&mut self, card_migrated: bool) {
```

Comment:
`card_migrated` can be optional value right ?

Reply:
no, card data will always be present (either raw card number or masked card number).
in case of raw card number, card_migrated is set to true.
in case of  masked card number, card_migrated will be set to false.